### PR TITLE
Generate Interfaces

### PR DESCRIPTION
**Pull request by ericanderson**

### CODE REVIEW

File: examples/basic/cli/src/OntologyType.ts [hunk 1]
```diff
@@ -17,12 +17,6 @@
...
-// This little trick makes it easier for me to see
-// what is happening in the types when I hover.
-// We need to actually generate some nice types here too
-type CapturedOntologyType = typeof Ontology;
-export interface OntologyType extends CapturedOntologyType {}
-
 // Demo for if you want concrete types
-export interface Employee extends OsdkObjectFrom<"Employee", OntologyType> {}
-export interface Todo extends OsdkObjectFrom<"Todo", OntologyType> {}
+export interface Employee extends OsdkObjectFrom<"Employee", Ontology> {}
```
Comment (ericanderson) [inline examples/basic/cli/src/OntologyType.ts:21]: Now that the Ontology file does the capture and type export this wasnt needed and confused people

File: examples/basic/sdk/ontology.json [hunk 2]
```diff
@@ -598,5 +598,38 @@
       "rid": "query.rid2",
       "version": "0.1.2"
     }
+  },
+  "__UNSTABLE_interfaceTypes": {
```
Comment (ericanderson) [inline examples/basic/sdk/ontology.json:602]: The backend doesnt actually return this right now, so I just hacked something together.

File: packages/generator/src/shared/sanitizeMetadata.ts [hunk 3]
```diff
@@ -14,11 +14,31 @@
...
  */
 
-import type { WireOntologyDefinition } from "../WireOntologyDefinition";
+import type {
+  __UNSABLE_WireOntologyDefinitionV2,
+  WireOntologyDefinition,
+} from "../WireOntologyDefinition";
 
+/**
+ * @internal
```
Comment (ericanderson) [inline packages/generator/src/shared/sanitizeMetadata.ts:23]: We should start littering this on all functions

File: packages/generator/src/WireOntologyDefinition.ts [hunk 4]
```diff
@@ -15,7 +15,19 @@
...
 
 import type * as gateway from "@osdk/gateway";
+import type { ObjectTypeV2 } from "@osdk/gateway/types";
 
 export interface WireOntologyDefinition
   extends gateway.components.OntologyFullMetadata
+{
+}
+
+export interface __UNSABLE_InterfaceType
```
Comment (mfedderly) [inline packages/generator/src/WireOntologyDefinition.ts]:
```suggestion
export interface __UNSTABLE_InterfaceType
```

^ this but do a global find and replace everywhere that's marked `UNSABLE`

File: examples/basic/sdk/src/generatedNoCheck/Ontology.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import type { OntologyDefinition } from '@osdk/api';
 import { OntologyMetadata } from './OntologyMetadata.js';
 import * as Actions from './ontology/actions/index.js';
-import * as Objects from './ontology/objects/index.js';
+import * as Interfaces from './ontology/interfaces.js';
+import * as Objects from './ontology/objects.js';
```
[the remark this answers is not in the document]
Comment (mfedderly) [reply examples/basic/sdk/src/generatedNoCheck/Ontology.ts:5]: I think this is a backcompat break though.

[the remark this answers is not in the document]
Comment (ericanderson) [reply examples/basic/sdk/src/generatedNoCheck/Ontology.ts:5]: this is for v2 only

**Pull request merged**